misc/logger/src/linkedList.ts

```diff
@@ -0,0 +1,110 @@
+class LogListNode {
+  private nodeValue: string;
+  private sizeInBytes: number;
+  public next: LogListNode | null;
+
+  constructor(value: string) {
+    this.nodeValue = value;
+    this.sizeInBytes = new TextEncoder().encode(this.nodeValue).length;
+    this.next = null;
+  }
+
+  public get value() {
+    return this.nodeValue;
+  }
+
+  public get size() {
+    return this.sizeInBytes;
+  }
+}
+
+export default class LogLinkedList {
+  private lengthInNodes: number;
+  private sizeInBytes: number;
+  private head: LogListNode | null;
+  private tail: LogListNode | null;
+  private maxSizeInBytes: number;
+
+  constructor(maxSizeInBytes: number) {
+    this.head = null;
+    this.tail = null;
+    this.lengthInNodes = 0;
+    this.maxSizeInBytes = maxSizeInBytes;
+    this.sizeInBytes = 0;
+  }
+
+  public append(value: string): void {
+    const newNode = new LogListNode(value);
+
+    while(this.size + newNode.size > this.maxSizeInBytes) {
```

Comment:
Should this be

```suggestion
    while(this.sizeInBytes + newNode.size > this.maxSizeInBytes) {
```

Reply:
no it's the same thing, `size` is a getter for `this.sizeInBytes`

Reply:
Gross but ok